### PR TITLE
Add org.shadowsocks.qt5client

### PR DESCRIPTION
**Pull request by wsgalaxy**
Shadowsocks-Qt5 is a native and cross-platform shadowsocks GUI client with advanced features.

I am not the developer of shadowsocks-qt5 , but It seems that upstream has no interest to provide a flatpak version: https://github.com/shadowsocks/shadowsocks-qt5/issues/345

see https://github.com/shadowsocks/shadowsocks-qt5

### CODE REVIEW

**Comment by wsgalaxy** [comment]
bot, build org.shadowsocks.qt5client

File: org.shadowsocks.qt5client.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      Shadowsocks-Qt5 is a native and cross-platform shadowsocks GUI client with advanced features.
+    </p>
+  </description>
+  <launchable type="desktop-id">org.shadowsocks.qt5client.desktop</launchable>
+  <categories>
+    <category>Network</category>
+  </categories>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://music.163.com/style/web2/img/down/b2.jpg</image>
```
Comment (AsciiWolf) [inline org.shadowsocks.qt5client.appdata.xml]: I don't think that this is a correct screenshot. :)

Comment (wsgalaxy) [reply org.shadowsocks.qt5client.appdata.xml]: It is my fault, I am looking a usable screenshot for shadowsocks-qt5.

Comment (AsciiWolf) [reply org.shadowsocks.qt5client.appdata.xml]: I see that you included a local screenshot in this PR. You can use `https://raw.githubusercontent.com/flathub/org.shadowsocks.qt5client/master/screenshot.png` as the screenshot url.

File: org.shadowsocks.qt5client.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          "url": "https://github.com/fukuchi/libqrencode.git",
+          "tag": "v4.0.2",
+          "commit": "59ee597f913fcfda7a010a6e106fbee2595f68e4"
+        }
+      ]
+    },
+    
+    {
+      "name": "libzbar",
+      "buildsystem": "autotools",
```
Comment (AsciiWolf) [inline org.shadowsocks.qt5client.json]: Try adding `"rm-configure": true,` here.

**Comment by wsgalaxy** [comment]
bot, build org.shadowsocks.qt5client

**Comment by AsciiWolf** [comment]
bot, build org.shadowsocks.qt5client

**Comment by wsgalaxy** [comment]
@AsciiWolf I can't understand the error message.

before adding `"rm-configure": true`:
```
Building module libzbar in /srv/buildbot/worker/build-x86_64/build/.flatpak-builder/build/libzbar-1
========================================================================
]2;flatpak-builder: Building libzbarFB: Running: tar xf /srv/buildbot/worker/build-x86_64/build/.flatpak-builder/downloads/dfcc940a42b2630d1aea0dcbb815fff85ca8b3b6534805986de5bed5aaf4cfcd/zbar-0.20.1.tar.gz --no-same-owner --strip-components=1 -z
]2;flatpak-builder: Cleanup libzbarError: module libzbar: Can't find autogen, autogen.sh or bootstrap
FB: unmounting rofiles-fuse /srv/buildbot/worker/build-x86_64/build/.flatpak-builder/rofiles/rofiles-gcoYKk
program finished with exit code 1
elapsedTime=978.053192
```
after adding `"rm-configure": true`:
```
Building module libzbar in /srv/buildbot/worker/build-x86_64-6/build/.flatpak-builder/build/libzbar-1
========================================================================
]2;flatpak-builder: Building libzbarFB: Running: tar xf /srv/buildbot/worker/build-x86_64-6/build/.flatpak-builder/downloads/dfcc940a42b2630d1aea0dcbb815fff85ca8b3b6534805986de5bed5aaf4cfcd/zbar-0.20.1.tar.gz --no-same-owner --strip-components=1 -z
]2;flatpak-builder: Cleanup libzbarError: module libzbar: Error removing file: No such file or directory
FB: unmounting rofiles-fuse /srv/buildbot/worker/build-x86_64-6/build/.flatpak-builder/rofiles/rofiles-jUOws7
program finished with exit code 1
elapsedTime=1106.327334
```

Shoud we  add autogen.sh for libzbar?

**Comment by AsciiWolf** [comment]
Yeah, this error message is weird. :-/ Maybe try using `"no-autogen": true,` instead.

**Comment by wsgalaxy** [comment]
bot, build org.shadowsocks.qt5client

**Comment by wsgalaxy** [comment]
bot, build org.shadowsocks.qt5client

**Comment by AsciiWolf** [comment]
Hmm, to be honest, I am not sure what's wrong with the build. :-(

**Comment by wsgalaxy** [comment]
May be we should use "simple" build system for libzbar .

**Comment by wsgalaxy** [comment]
bot, build org.shadowsocks.qt5client

**Comment by wsgalaxy** [comment]
bot, build org.shadowsocks.qt5client

**Comment by wsgalaxy** [comment]
bot, build org.shadowsocks.qt5client

**Comment by wsgalaxy** [comment]
bot, build org.shadowsocks.qt5client

**Comment by wsgalaxy** [comment]
Work well for me now, we need more appdata.xml updates.

**Comment by wsgalaxy** [comment]
bot, build org.shadowsocks.qt5client

**Comment by AsciiWolf** [comment]
bot, build org.shadowsocks.qt5client

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.shadowsocks.qt5client

**Pull request closed without merging**